### PR TITLE
GRBL: Lots of changes for various NS related things

### PR DESCRIPTION
**Pull request by mdicicco**
Sorry for the big dump, this is a lot of overdue stuff.

Probably best to comment on individual commits since there is a lot of whitespace cleanup here too.

### CODE REVIEW

**Comment by penguinland** [comment]
So far, this looks really good. The only commits I haven't yet reviewed are:
- https://github.com/keyme/grbl/commit/c6524d2e095c77d1147782c1200d23191ff81200
- https://github.com/keyme/grbl/commit/4d162d1919f2b72b424d32d9edd5b5305b1ddda2
- https://github.com/keyme/grbl/commit/963535e86fe0870b3cf2f8b1489a700e5916f985

**Comment by mdicicco** [comment]
ha.  I think those commits are the bulk of the important stuff

**Comment by mdicicco** [comment]
Updated just now based on a change to use EEPROM to save/change the force sensor sensitivity value.  I tagged this as 3.2 because it will be important to know whether or now we have support for this on a kiosk.

**Comment by penguinland** [comment]
I haven't gotten a chance to look at the rest of this yet. Sorry about the delays. Anyone else is welcome to review in my stead.

File: cpu_map_keyme2560.h [hunk 1]
```diff
@@ -155,22 +155,43 @@
 #define PFD_MASK 0xFF //all 8 bits are used, in XYZC order
 #define SET_DECAY_MODE(axis,val) (val<<((axis*2)&3)) //2 bits
 
+#define NEW_BOARD
```
Comment (kborer) [inline cpu_map_keyme2560.h:158]: what happens when the new board is the old board?

File: cpu_map_keyme2560.h [hunk 2]
```diff
@@ -155,22 +155,43 @@
...
 #define FDBK_INT_vect  PCINT2_vect
 #define FDBK_PCMSK     PCMSK2 // Pin change interrupt register
-#define FDBK_MASK ((1<<Z_ENC_IDX_BIT)|(1<<Z_ENC_CHA_BIT)|(1<<Z_ENC_CHB_BIT) |(1<<ALIGN_SENSE_BIT)  )
-
-#define NEW_BOARD
-#ifdef NEW_BOARD
+#define FDBK_MASK (1<<ALIGN_SENSE_BIT)
+
+// XY or CG Current Control outputs
+// TODO: remove this functionaltiy on rev5 and beyond in favor of PWM pin
```
Comment (kborer) [inline cpu_map_keyme2560.h]: functionality

File: cpu_map_keyme2560.h [hunk 3]
```diff
@@ -155,22 +155,43 @@
...
+#define CCTRL_DDR DDRB
+#define CCTRL_PORT PORTB
+#define CCTRL_CG_BIT 5
+#define CCTRL_XY_BIT 4
+
+// Force Sensor Sensitivity Control output
+// TODO: remove this functionaltiy on rev5 and beyond in favor of PWM pin
```
Comment (kborer) [inline cpu_map_keyme2560.h]: here too

File: cpu_map_keyme2560.h [hunk 4]
```diff
@@ -178,13 +199,45 @@
...
+#define FDBK_PIN PINK
+#define ALIGN_SENSE_BIT 3  //This is also PROBE_BIT
+#define Z_ENC_IDX_BIT 4
+#define Z_ENC_CHA_BIT 5
+#define Z_ENC_CHB_BIT 6
+
+#define FDBK_INT       PCIE2  // Pin change interrupt enable pin
+#define FDBK_INT_vect  PCINT2_vect
+#define FDBK_PCMSK     PCMSK2 // Pin change interrupt register
+#define FDBK_MASK ((1<<Z_ENC_IDX_BIT)|(1<<Z_ENC_CHA_BIT)|(1<<Z_ENC_CHB_BIT) |(1<<ALIGN_SENSE_BIT)  )
```
Comment (kborer) [inline cpu_map_keyme2560.h]: inconsistent spacing

File: report.c [hunk 5]
```diff
@@ -381,30 +381,80 @@ void report_counters()
...
-  printInteger((volts>>Y_MVOLT_BIT) & 1);
-  printPgmString(PSTR(","));
-  printInteger((volts>>Z_MVOLT_BIT) & 1);
-  printPgmString(PSTR(","));
-  printInteger((volts>>C_MVOLT_BIT) & 1);
+
+  // This block should work to read the 4 analogs connected
+  // to the 24V feed lines.  It does work once on each new
+  // instance of a session, but then it goes to nonsense.
+  // Better to just skip it, or we may mess up the
```
Comment (kborer) [inline report.c:393]: better to just get rid of the code if it is not needed

File: cpu_map_keyme2560.h (hunk 1 above)
Comment (mdicicco) [inline cpu_map_keyme2560.h:158]: NEW_NEW_
NEW_NEW_NEW_
NEW_X4_
NEW_FOR_REAL_THIS_TIME_
BEST_NEW_
FINAL_BOARD_
FINAL_W_UPDATE_
FINAL_2_

File: stepper.c [hunk 6]
```diff
@@ -519,36 +519,149 @@ void st_reset()
...
+  TCCR3C = 0;
+
+  // Set all three pins as OUTPUT
+  PWM_OUT_DDR |= 1<<PWM_OUT_FSENSE_BIT;
+  PWM_OUT_DDR |= 1<<PWM_OUT_XY_CTRL_BIT;
+  PWM_OUT_DDR |= 1<<PWM_OUT_CG_CTRL_BIT;
+
+  // Set OLD PWM Sensor Drive as INPUT so it's not driving anything
+  // TODO: remove this as un-needed on Rev5 boards
+  FSCTRL_DDR |= 0<<FSCTRL_BIT;
```
Comment (kborer) [inline stepper.c:601]: if it's not needed on rev5, perhaps wrap it in a board-version conditional

File: stepper.c [hunk 7]
```diff
@@ -519,36 +519,149 @@ void st_reset()
...
+
+  // With the comparators and timers set up, use OCR3xL to set compare value
+  // In 8 bit mode, only the Low register is needed
+  OCR3AL = 51;  // 1V
+  OCR3BL = settings.force_sensor_level; // 1.5V default
+  OCR3CL = 153; // 3V
+
+  /*
+  // THIS WAS TEST CODE TO TRY TO DRIVE A SOFTWARE PWM
+  // Saved for posterity and maybe future use
```
Comment (kborer) [inline stepper.c:611]: save it in a separate file

File: stepper.c (hunk 6 above)
Comment (mdicicco) [inline stepper.c:601]: In board conditional is hard.  I know the IO board can see that voltage divider.  I don't know if we have it connected to an AIN on motion.

File: report.c (hunk 5 above)
Comment (mdicicco) [inline report.c:393]: I'd like to keep this.  We want to enable reading these bits in the future.

File: stepper.c (hunk 7 above)
Comment (mdicicco) [inline stepper.c:611]: A separate file just for this code?  The location within this particular spot in the code is somewhat important if we ever do need to turn it on.

Maybe it's okay to just delete it, given the fact that we're making the hardware change.

**Comment by mdicicco** [comment]
I want this merge to help me work remotely from benchmark next week.  If there are no objections, I'll go ahead and make the merge and post the hex to the kiosk repo.

**Comment by kborer** [comment]
lgtm

**Comment by kborer** [comment]
squash

**Comment by mdicicco** [comment]
squashed

**Pull request merged**